hooks/cron.ts

```diff
@@ -0,0 +1,43 @@
+import { ofetch, $Fetch } from "ofetch";
+import { CronJob } from "cron";
+import { publicRuntimeConfig } from "../nuxt.config";
+import { getNextQueuedScene, popNextQueuedScene } from "../utils/apis";
+
+async function dailyFeatureSetup(backendCall: $Fetch) {
+  const now = new Date();
+  const tomorrow = new Date(now);
+  tomorrow.setDate(tomorrow.getDate() + 1);
+  const features = await getFeaturesInRange(backendCall, now.getTime(), tomorrow.getTime());
+
+  if (features.length === 0) {
+    features.forEach(feature => {
+      const date = new Date(feature.feature_time);
+      const featureJob = new CronJob(date, () => {
+          miscUpdateTimeline(backendCall, feature.scene.id);
+        },
+        null, true, "America/New_York");
+      });
+  } else {
+    const nextInQueue = await getNextQueuedScene(backendCall);
+    await miscUpdateTimeline(backendCall, nextInQueue.id);
+    popNextQueuedScene(backendCall);
+  }
+}
+
+
+export default function schedulerHook() {
+  const backendCall = ofetch.create({
+    baseURL: publicRuntimeConfig.apiUrl,
+    headers: {
+      "Accept": "application/json"
+    },
+  });
```

Comment:
So, in order for this functionality to work in a secure production system, the backend will need to have some kind of way to recognize that this API call is coming from the frontend server and therefore grant permissions to make changes that we wouldn't allow for unauthenticated requests coming off the open internet. As it stands, we don't have any such mechanism, and I believe that the requests in this module will be made with no auth info and ones requiring privileges should fail.

I'm sure that there's a standard way to make it so that the frontend can make "special" requests, but I'm not actually sure what it might be. The requests will be coming over an Azure VPN rather than the open internet, and it seems better to grant permissions based on that rather than, say, giving the frontend some secret magic token. Maybe you just hardcode the backend to notice requests coming from the appropriate VPN IP subnet and trust that traffic from the open internet is cannot have those IPs?